src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -232,6 +238,12 @@ private async Task<bool> TryPopulateNewJobFields(CancellationToken cancellationT
                 }
             }
 
+            // save the list of search parameters to the reindexjob document
+            foreach (var searchParams in notYetIndexedParams.Select(p => p.Url.OriginalString))
+            {
```

Comment:
We should probably do a sanity check here.

                if (_reindexJobRecord.SearchParams.Contains(searchParams))
                {
                    continue;
                }